### PR TITLE
Eliminated method useIsEqualToInsteadOfIsSameAs() from AbstractSeqTes…

### PR DESCRIPTION
**Pull request by NataliiaPrivezentseva**
…t class and prepared code for elimination this method from related classes: ArrayTest, HashSetTest, ListTest, QueueTest, StreamTest and VectorTest

### CODE REVIEW

File: vavr/src/test/java/io/vavr/collection/AbstractSeqTest.java [hunk 1]
```diff
@@ -140,14 +140,14 @@ public void shouldNarrowSeq() {
 
     @Test
     public void shouldAppendElementToNil() {
-        final Seq<Integer> actual = this.<Integer> empty().append(1);
+        final Seq<Integer> actual = this.<Integer>empty().append(1);
```
Comment (valery1707) [inline vavr/src/test/java/io/vavr/collection/AbstractSeqTest.java]: Was it necessary to change the formatting of the code?
Formatting changes a hide the real changes and not necessary as I think.

File: vavr/src/test/java/io/vavr/collection/AbstractSeqTest.java [hunk 2]
```diff
@@ -1346,27 +1335,14 @@ public void shouldReturnSameSeqWhenEmptyRemoveAllNonEmpty() {
 
     @Test
     public void shouldRemoveExistingElements() {
-        assertThat(of(1, 2, 3).removeAll(i -> i == 1)).isEqualTo(of(2, 3));
```
Comment (valery1707) [inline vavr/src/test/java/io/vavr/collection/AbstractSeqTest.java]: Why you remove this tests?

File: vavr/src/test/java/io/vavr/collection/StreamTest.java [hunk 3]
```diff
@@ -198,11 +203,73 @@
...
     protected boolean useIsEqualToInsteadOfIsSameAs() {
         return true;
     }
 
+    @Test
+    @Override
+    public void shouldRemoveNonExistingElement() {
+        final Seq<Integer> t = of(1, 2, 3);
+        assertThat(t.remove(4)).isEqualTo(t);
+        assertNotSame(t, t.remove(4));
```
Comment (valery1707) [inline vavr/src/test/java/io/vavr/collection/StreamTest.java]: All current test is used in form `assertThat(actual).testMethod(expected);` and more - `assertThat(actual)` is support chaining: `assertThat(t.remove(4)).isEqualTo(t).isNotSameAs(t);`

File: vavr/src/test/java/io/vavr/collection/StreamTest.java [hunk 4]
```diff
@@ -198,11 +203,73 @@
...
+        final Seq<Integer> t = of(1, 2, 3);
+        assertThat(t.removeAll(of(4, 5))).isEqualTo(t);
+        assertNotSame(t, t.removeAll(of(4, 5)));
+    }
+
+    @Test
+    @Override
+    public void shouldRemoveExistingElements() {
+        final Seq<Integer> seq = of(1, 2, 3);
+        assertThat(seq.removeAll(i -> i == 1)).isEqualTo(of(2, 3));
```
Comment (valery1707) [inline vavr/src/test/java/io/vavr/collection/StreamTest.java]: `removeAll(Predicate)` is deprecated now - you must use `reject(Predicate)` or add annotation `@SuppressWarnings("deprecation")` to the test method

File: vavr/src/test/java/io/vavr/collection/AbstractSeqTest.java (hunk 1 above)
Comment (NataliiaPrivezentseva) [reply vavr/src/test/java/io/vavr/collection/AbstractSeqTest.java]: It was not done on purpose, I used Ctrl + Alt + L and these changes were performed

File: vavr/src/test/java/io/vavr/collection/StreamTest.java (hunk 3 above)
Comment (NataliiaPrivezentseva) [reply vavr/src/test/java/io/vavr/collection/StreamTest.java]: I didn't khow that! Thanks! I'll change my code

File: vavr/src/test/java/io/vavr/collection/StreamTest.java (hunk 4 above)
Comment (NataliiaPrivezentseva) [reply vavr/src/test/java/io/vavr/collection/StreamTest.java]: Thank you, I'll change this

File: vavr/src/test/java/io/vavr/collection/AbstractSeqTest.java (hunk 1 above)
Comment (NataliiaPrivezentseva) [reply vavr/src/test/java/io/vavr/collection/AbstractSeqTest.java]: @valery1707 do you think I should change it back?

Comment (valery1707) [reply vavr/src/test/java/io/vavr/collection/AbstractSeqTest.java]: @NataliiaPrivezentseva, I think that you should not change formatting of lines that you don't really change.

File: vavr/src/test/java/io/vavr/collection/AbstractSeqTest.java [hunk 5]
```diff
@@ -1217,11 +1221,7 @@ public void shouldNotRemoveDuplicateElement() {
     @Test
     public void shouldRemoveNonExistingElement() {
         final Seq<Integer> t = of(1, 2, 3);
-        if (useIsEqualToInsteadOfIsSameAs()) {
-            assertThat(t.remove(4)).isEqualTo(t);
-        } else {
-            assertThat(t.remove(4)).isSameAs(t);
-        }
+        assertThat(t.remove(4)).isSameAs(t);
```
Comment (danieldietrich) [inline vavr/src/test/java/io/vavr/collection/AbstractSeqTest.java]: Great to see that we can get rid of it in our Seq tests!

**Comment by NataliiaPrivezentseva** [comment]
@danieldietrich, @valery1707 Could you please look at changes I've made?

File: vavr/src/test/java/io/vavr/collection/AbstractSeqTest.java [hunk 6]
```diff
@@ -1328,22 +1328,20 @@ public void shouldReturnSameSeqWhenEmptyRemoveAllNonEmpty() {
...
-    @SuppressWarnings("deprecation")
     @Test
     public void shouldRemoveExistingElements() {
         final Seq<Integer> seq = of(1, 2, 3);
-        assertThat(seq.removeAll(i -> i == 1)).isEqualTo(of(2, 3));
-        assertThat(seq.removeAll(i -> i == 2)).isEqualTo(of(1, 3));
-        assertThat(seq.removeAll(i -> i == 3)).isEqualTo(of(1, 2));
-        assertThat(seq.removeAll(ignore -> true)).isEmpty();
-        assertThat(seq.removeAll(ignore -> false)).isSameAs(seq);
+        assertThat(seq.reject(i -> i == 1)).isEqualTo(of(2, 3));
```
Comment (valery1707) [inline vavr/src/test/java/io/vavr/collection/AbstractSeqTest.java]: This test must use exactly `removeAll` method. It is deprecated, but must be tested too

File: vavr/src/test/java/io/vavr/collection/AbstractSeqTest.java [hunk 7]
```diff
@@ -1328,22 +1328,20 @@ public void shouldReturnSameSeqWhenEmptyRemoveAllNonEmpty() {
...
+        assertThat(seq.reject(ignore -> true)).isEmpty();
+        assertThat(seq.reject(ignore -> false)).isSameAs(seq);
     }
 
-    @SuppressWarnings("deprecation")
     @Test
     public void shouldRemoveNonExistingElements() {
-        assertThat(this.<Integer> empty().removeAll(i -> i == 0)).isSameAs(empty());
-        assertThat(of(1, 2, 3).removeAll(i -> i != 0)).isSameAs(empty());
+        assertThat(this.<Integer> empty().reject(i -> i == 0)).isSameAs(empty());
```
Comment (valery1707) [inline vavr/src/test/java/io/vavr/collection/AbstractSeqTest.java]: `removeAll`

File: vavr/src/test/java/io/vavr/collection/AbstractSeqTest.java [hunk 8]
```diff
@@ -1362,12 +1362,11 @@ public void shouldRemoveAllMatchedElementsFromNonNil() {
         assertThat(of(1, 2, 3, 4, 5, 6).removeAll(i -> i % 2 == 0)).isEqualTo(of(1, 3, 5));
     }
 
-    @SuppressWarnings("deprecation")
     @Test
     public void shouldNotRemoveAllNonMatchedElementsFromNonNil() {
         final Seq<Integer> t = of(1, 2, 3);
         final Predicate<Integer> isTooBig = i -> i >= 4;
-        assertThat(t.removeAll(isTooBig)).isSameAs(t);
+        assertThat(t.reject(isTooBig)).isSameAs(t);
```
Comment (valery1707) [inline vavr/src/test/java/io/vavr/collection/AbstractSeqTest.java]: `removeAll`

File: vavr/src/test/java/io/vavr/collection/StreamTest.java [hunk 9]
```diff
@@ -236,11 +236,11 @@ public void shouldNotRemoveAllNonExistingElementsFromNonNil() {
     @Override
     public void shouldRemoveExistingElements() {
         final Seq<Integer> seq = of(1, 2, 3);
-        assertThat(seq.removeAll(i -> i == 1)).isEqualTo(of(2, 3));
-        assertThat(seq.removeAll(i -> i == 2)).isEqualTo(of(1, 3));
-        assertThat(seq.removeAll(i -> i == 3)).isEqualTo(of(1, 2));
-        assertThat(seq.removeAll(ignore -> true)).isEmpty();
-        assertThat(seq.removeAll(ignore -> false)).isEqualTo(of(1, 2, 3)).isNotSameAs(of(1, 2, 3));
+        assertThat(seq.reject(i -> i == 1)).isEqualTo(of(2, 3));
```
Comment (valery1707) [inline vavr/src/test/java/io/vavr/collection/StreamTest.java]: If you test call to method `reject` you test method should include word `reject` but it include `remove`

**Comment by danieldietrich** [comment]
Many thanks, also to the reviewers! Good work.

**Pull request merged**